rust/core-lib/src/event_context.rs

```diff
@@ -414,6 +454,30 @@ impl<'a> EventContext<'a> {
                            ed.get_layers().get_merged(), first, last,
                            ed.is_pristine())
     }
+
+    fn request_hover(&mut self, request_id: usize, position: Option<Position>) {
+        
+        let position = position.or_else(|| self.view.borrow().get_caret_offset()
```

Comment:
I am confused. If Plugins specify their preference i.e Utf16 Line/Char or Utf8 or anything else, how will that be handled without enum?

Reply:
Currently, plugins only get sent utf8 offsets, and that isn't going to change soon.

If at some point in the future, we allow a plugin to specify it's preferred unit, the RPC it receives will still just be (probably) `hover { position: { line: 0, col: 3 }`, it will just know that col is in utf16 code units. We _could_ represent this internally as an enum, (and that might be best) but we might not send it as an enum to the plugin; we could switch internally and modify the json as needed. (I'm not sure this is a good idea, just a thought).